lib/filter/buddy.js

```diff
@@ -115,14 +115,18 @@ exports.update = function(json) {
     /**
      * Hit-all
      * Overrides the record materia with Zealot and adds the Zealot Attack property
+     * This also utilizes elementType and multiHit to determine the element type and number of hits.
      * Attack CAN be modified to do several things - hit more than once, etc.
      */
-  	if (config.get('buddy.attack.multiHit')) {
-	    var materia = recordMateria["ZEALOT"];
-	    var materiaatk = recordMateria["ZEALOTATK"];
-	    json[buddy].materias = materia;
-	    json[buddy].abilities.push(materiaatk);
-	  }
+    if (config.get('buddy.attack.hitAll')) {
+      var materia = recordMateria["ZEALOT"];
+      var materiaatk = recordMateria["ZEALOTATK"];
+      var elementid = config.get('buddy.attack.elementType');
+      json[buddy].materias = materia;
+      json[buddy].abilities.push(materiaatk);
+      json[buddy].abilities[json[buddy].abilities.length - 1].options.arg2 = config.get('buddy.attack.multiHit');
```

Comment:
Ah my bad, I think only objects are in reference.
```
var abilitiy = json[buddy].abilities[json[buddy].abilities.length - 1]
```
This should work, but maybe it is easier to just save the index somewhere, this would make the code small/cleaner